packages/services/f-feature-management/src/evaluateRestriction.js

```diff
@@ -0,0 +1,220 @@
+import { logger } from './logger';
+
+/**
+ * Checks that the restriction object has at most one operator specified
+ * @param {object} restriction. Throws if not.
+ */
+function verifyRestrictionObject (restriction) {
+    const operatorsSpecified = ['and', 'or', 'eq', 'neq', 'in', 'nin', 'gt', 'gte', 'lt', 'lte']
+    .filter(operator => restriction[operator]);
+
+    if (operatorsSpecified.length > 1) {
+        throw new Error(`${operatorsSpecified.length} operators specified`, operatorsSpecified);
+    }
+}
+
+/**
+ * Utility function to return true if predicate passes for every item in the array.
+ * @param {array} array
+ * @param {function} predicate - map from the array items
+ * @returns Boolean
+ */
+function all (array, predicate) {
+    const passes = array.filter(i => predicate(i));
```

Comment:
There's already a built-in JS array method called `every` that you could try using?

Reply:
TIL